App/Containers/TextileManager.js

```diff
@@ -52,13 +52,22 @@ class TextileManager extends React.PureComponent {
     this.props.appStateChange(this.props.currentAppState, newAppState)
   }
 
+  handleUploadComplete (event) {
+    const {responseCode} = event
+    if (responseCode >= 200 && responseCode < 300) {
+      this.props.uploadComplete(event)
+    } else {
+      this.props.uploadError({...event, error: 'Response code: ' + responseCode})
+    }
+  }
+
   async setup () {
     // await PushNotificationIOS.requestPermissions()
     AppState.addEventListener('change', this.handleNewAppState.bind(this))
     navigator.geolocation.watchPosition(() => this.props.locationUpdate(), null, { useSignificantChanges: true })
     this.progressSubscription = Upload.addListener('progress', null, this.props.uploadProgress)
-    this.completionSubscription = Upload.addListener('completed', null, this.props.uploadComplete)
-    this.cancelledSubscription = Upload.addListener('cancelled', null, this.props.uploadCancelled)
+    this.completionSubscription = Upload.addListener('completed', null, this.handleUploadComplete.bind(this))
```

Comment:
Delegate handling the upload complete event to the function above...

We were previously assuming all 'complete' uploads are successful, but 'complete' uploads include those with non 200-299 http response codes. So that function inspects the response code and calls the appropriate handler, `uploadComplete` or `uploadError`